herbert_core/classes/MPIFramework/@JobDispatcher/private/split_tasks_.m

```diff
@@ -7,105 +7,115 @@
+                init_messages{i} = InitMessage(common_par,...
+                                               param,return_outputs,1);
+            else
+                init_messages{i} = InitMessage(common_par,...
+                                               loop_par(worker_par_list{i}),return_outputs,1);
+            end
         else
+            ind = worker_par_list{i};
             init_messages{i} = InitMessage(common_par,...
-                loop_par(worker_par_list{i}),return_outputs,1);
+                                           ind(2),return_outputs,ind(1));
         end
-    else
-        ind = worker_par_list{i};
-        init_messages{i} = InitMessage(common_par,...
-            ind(2),return_outputs,ind(1));
     end
 end
 
 function struct_par = split_struct(cell_data,field_names,array_cur)
-n_fields = numel(field_names);
-struct_par  = cell(n_fields,1);
-for i=1:n_fields
-    celd =  cell_data{i};
-    if size(celd,1)>    size(celd,2)
-        celd = celd';
-    end
-    if iscell(celd)
-        struct_par{i} = celd(array_cur);
-    elseif array_cur == 1
-        struct_par{i} = celd;
-    else
-        error('JOB_DISPATCHER:invalid_argument',...
-            'unsupported combination of cell data and cell indexes');
+
+    n_fields = numel(field_names);
+    struct_par  = cell(n_fields,1);
+
+    for i=1:n_fields
+        celd = cell_data{i};
+
+        if size(celd,1) > size(celd,2)
+            celd = celd';
+        end
+
+        if iscell(celd)
+            struct_par{i} = celd(array_cur);
+        elseif array_cur == 1
+            struct_par{i} = celd;
+        else
+            error('JOB_DISPATCHER:invalid_argument',...
+                  'unsupported combination of cell data and cell indices');
+        end
     end
-end
-struct_par = cell2struct(struct_par,field_names);
 
+    struct_par = cell2struct(struct_par,field_names);
+end
 
-function [n_workers,worker_par_list,is_list] = tasks_indexes(job_param_list,n_workers)
-% get subtasks indexes, dividing input parameter list between defined number of workers.
+function [n_workers,worker_par_list,is_list] = tasks_indices(job_param_list,n_workers)
+% get subtasks indices, dividing input parameter list between defined number of workers.
 %
-if iscell(job_param_list) % the tasks are described by cellarray
-    n_tasks = numel(job_param_list);
-    is_list = true;
-elseif isstruct(job_param_list) % array of structures
-    fn = fieldnames(job_param_list);
-    if ~isempty(fn)
-        n_tasks = numel(job_param_list.(fn{1}));
+    if ~isscalar(job_param_list) % the tasks are described by array
+        n_tasks = numel(job_param_list);
         is_list = true;
-    else
-        error('JOB_DISPATCHER:invalid_argument',...
-            ' job loop parameters contains empty structure');
+
+    elseif isstruct(job_param_list) % array of structures
+
+        fn = fieldnames(job_param_list);
+        if ~isempty(fn)
+            n_tasks = numel(job_param_list.(fn{1}));
```

Comment:
This is assuming something about  that the contents of the "first" field in the struct.
The documentation for `fieldnames` doesn't offer any guarantee what order fields will be returned.